### DISCUSSION TITLE
How to Inject Controller constructor with authorization

### DISCUSSION THREAD

**Discussion by hendra1** [adonisjs/core#1299] [Help] [answered]
Hi All...

I wonder how can I make an injection to the controller.
For example like this one:

`
class UserController {
  constructor() {
    this.UserService = new UserService(auth)
  }
}
`

any suggests?

Thanks in advance

**Comment by iammrsea** [comment]
I am not sure you can. Where do you intend to get the **auth** object you are passing to your UserService from? Instead, I would suggest you should abstract the logic in your UserService that needs the **auth** object into a middleware.

**Comment by Lazzzer** [comment]
I think it would be better to use the auth package provided by Adonis. You will have ``auth`` available for your controller via its context. Docs for [V4](https://adonisjs.com/docs/4.1/authentication) and [V5](https://preview.adonisjs.com/guides/auth/introduction).

**Comment by hendra1** [comment, marked as answer]
Hi...
I found the solution...
in the middleware 
I use the ioc.bind to return the new UserService.

and then in UserController inside the constructor : 
`
this.userService = use('UserService')
`
